app/services/appointment_notification_service.rb

```diff
@@ -21,11 +20,13 @@ def send_after_missed_visit
     eligible_appointments.each do |appointment|
       next if appointment.previously_communicated_via?(communication_type)
 
-      AppointmentNotification::Worker.perform_at(schedule_at, appointment.id, communication_type)
+      next_messaging_time = Communication.next_messaging_time
```

Comment:
This is the meat of the refactoring done in this PR. This service fetches the "next message time" from the Communication model directly, instead of receiving it as an arg through a few layers of application code.